api/tests/utils/filter-items.test.ts

```diff
@@ -0,0 +1,63 @@
+import { filterItems } from '../../src/utils/filter-items';
+
+const items = [
+	{
+		role: '9bc9fea0-f761-4107-bfb7-b3d06c125e98',
+		permissions: {},
+		validation: null,
+		presets: null,
+		fields: ['*'],
+		system: true,
+		collection: 'directus_settings',
+		action: 'read',
+	},
+	{
+		role: '9bc9fea0-f761-4107-bfb7-b3d06c125e98',
+		permissions: {
+			user: {
+				_eq: '$CURRENT_USER',
+			},
+		},
+		validation: null,
+		presets: null,
+		fields: ['*'],
+		system: true,
+		collection: 'directus_presets',
+		action: 'delete',
+	},
+];
+
+describe('filter items', () => {
+	test('return items when no filter', () => {
+		const result = filterItems(items, undefined);
+		expect(result).toStrictEqual(items);
+	});
+
+	test('return items when empty filter used', () => {
+		const result = filterItems(items, {});
+		expect(result).toStrictEqual(items);
+	});
+
+	test('return filtered items when nested empty filter used', () => {
+		const result = filterItems(items, {
+			_or: [
```

Comment:
The test should be for `_and` as `{}` skips filtering in `_or`.

```suggestion
			_and: [
```

Reply:
Ah very true 🤦‍♂️